tests/Integration/FrameworkIntegrationTestCase.php

```diff
@@ -44,6 +45,11 @@ protected function setUp(): void
         @unlink($databasePath);
         copy($cleanDatabasePath, $databasePath);
 
+        $filename = __DIR__ . '/../Fixtures/Config/database.php';
+        if (! file_exists($filename)) {
```

Comment:
Do we want an additional filesystem I/O operation on every test? Wouldn't it fail somewhere else if we forgot to set up te database?

Reply:
We could just let it fail hard with stack trace and all. But I didn't think that was very user/developer friendly, and it would also be harder to see why the tests wouldn't run.

I realise that this also throws an exception, however the stack trace would be significantly smaller and there's only one point of origin. You wouldn't need to follow the trace to where it originated. Which again is not impossible, but it would take longer. Or at least that was my reasoning behind adding this.

We could also place it higher, above the unlinking or even before the `appConfig`.

Reply:
additionally we could also add the following. to reduce file operations with other database drivers.

```php
$driver = $this->container->get(DatabaseDriver::class);
if ($driver->dialect()->value === DatabaseDialect::SQLITE->value) {
    $databasePath = __DIR__ . '/../Fixtures/database.sqlite';
    $cleanDatabasePath = __DIR__ . '/../Fixtures/database-clean.sqlite';

    @unlink(DiscoveryDiscovery::CACHE_PATH);
    @unlink($databasePath);
    copy($cleanDatabasePath, $databasePath);
}
```

Reply:
Yeah I like that :)